platform/android/src/geojson/feature_collection.cpp

```diff
@@ -0,0 +1,40 @@
+#include "feature_collection.hpp"
+
+#include "feature.hpp"
+
+namespace mbgl {
+namespace android {
+namespace geojson {
+
+mbgl::FeatureCollection FeatureCollection::convert(jni::JNIEnv& env, jni::Object<FeatureCollection> jCollection) {
+    auto jFeatureList = FeatureCollection::getFeatures(env, jCollection);
+    auto jFeatures = java::util::List::toArray<Feature>(env, jFeatureList);
```

Comment:
No, haven't benchmarked this. Curious if a method call (though collection class methods) has more overhead than a memory access through array access.

Options 2 and 3 are not applicable here though, region access is not available for objects, only primitives as far as I know.